Modules/_sqlite/connection.c

```diff
@@ -1442,6 +1457,169 @@ pysqlite_connection_iterdump(pysqlite_Connection* self, PyObject* args)
     return retval;
 }
 
+#ifdef HAVE_BACKUP_API
+static PyObject *
+pysqlite_connection_backup(pysqlite_Connection *self, PyObject *args, PyObject *kwds)
+{
+    PyObject *target = NULL;
+    const char *filename;
+    int pages = -1;
+    PyObject *progress = Py_None;
+    char *name = "main";
+    int rc;
+    int cberr = 0;
+    double sleep_secs = 0.250;
+    sqlite3 *bckconn;
+    sqlite3_backup *bckhandle;
+    static char *keywords[] = {"target", "pages", "progress", "name", "sleep", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|$iOsd:backup", keywords,
+                                     &target, &pages, &progress, &name, &sleep_secs)) {
+        return NULL;
+    }
+
+    if (!PyUnicode_Check(target) && !PyObject_TypeCheck(target, &pysqlite_ConnectionType)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "target argument must be either a string or a Connection instance");
+        return NULL;
+    }
+
+    if (progress != Py_None && !PyCallable_Check(progress)) {
```

Comment:
Sorry, I missed this. We prefer ``PyNone_Check()`` these days.

Reply:
Uhm, really? The only reference I can find is in `Doc/c-api/none.rst` and it says:
```
Note that the :c:type:`PyTypeObject` for ``None`` is not directly exposed in the
Python/C API.  Since ``None`` is a singleton, testing for object identity (using
``==`` in C) is sufficient. There is no :c:func:`PyNone_Check` function for the
same reason.
```
Am I wrong?

Reply:
Nope, you are right and I got fooled by some legacy code base. I have been banging my head against old C code for the last weeks. The code base has ``#define PyNone_Check(o) ((o) == Py_None)``.

Scratch my comment.